### PR TITLE
add e2e test and github CI

### PR DESCRIPTION
**Pull request by ryanzhang-oss**

### CODE REVIEW

**Review by wonderflow — APPROVED** [review]
LGTM

File: e2e-test/suite_test.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	}
+	close(done)
+}, 300)
+
+var _ = AfterSuite(func() {
+	By("Tearing down the test environment")
+})
+
+// Match the error to be already exist
+type AlreadyExistMatcher struct {
```
Comment (hongchaodeng) [inline e2e-test/suite_test.go:68]: How does this Matcher works?

Comment (ryanzhang-oss) [reply e2e-test/suite_test.go:68]: I am not entirely sure about the question. This matcher is used to improve test reliability following the best practice in distributed computing.

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**